fdbserver/storageserver.actor.cpp

```diff
@@ -9678,7 +9678,18 @@ ACTOR Future<Void> updateStorage(StorageServer* data) {
 				wait(delay(endTime - now(), TaskPriority::UpdateStorage));
 			}
 		}
-		wait(data->desiredOldestVersion.whenAtLeast(data->storageVersion() + 1));
+
+		// We might arrive here if after the last commit the FetchKeys budget was used up prior to the
+		// STORAGE_COMMIT_INTERVAL delay.  In this case, we want to wait some time for mutations to arrive
+		// so we do not fall behind on the mutations stream due to FetchKeys activity.  However, this delay
+		// should be short because
+		//   - In an idle cluster, the next commit version could arrive hundreds of ms from now
+		//   - In a non-idle cluster, the update() actor would already have update requests outstanding
+		//     even during the last updateStorage() commit cycle so the next versions should already be here.
+		Future<Void> maxWait = data->fetchKeysBudgetUsed.get()
+		                           ? delay(SERVER_KNOBS->STORAGE_COMMIT_MAX_WAIT_AFTER_FETCH_BUDGET_USED)
+		                           : Never();
+		wait(data->desiredOldestVersion.whenAtLeast(data->storageVersion() + 1) || maxWait);
```

Comment:
@etschannen The more I think about this, I think this delay can just be `delay(0, TaskPriority::UpdateStorage)` because of point 2 in the command.